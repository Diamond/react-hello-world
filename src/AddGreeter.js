import React, { Component } from 'react';
import './AddGreeter.css';

class AddGreeter extends Component {
  constructor(props) {
    super(props);
    this.state = { greetingName: '' };
    this.handleUpdate = this.handleUpdate.bind(this);
    this.addGreeting = this.addGreeting.bind(this);
  }
  handleUpdate(event) {
    // Use this.setState.then?
    this.setState({ greetingName: event.target.value });
  }
  addGreeting() {
    this.props.addGreeting(this.state.greetingName);
    this.setState({ greetingName: '' });
  }
  render() {
    return (
      <div className="AddGreeter">
        <input
          type="text"
          onChange={this.handleUpdate}
          value={this.state.greetingName}
        />
        &nbsp;&nbsp;
        <button onClick={this.addGreeting}>Add</button>
      </div>
    );
  }
}

export default AddGreeter;
