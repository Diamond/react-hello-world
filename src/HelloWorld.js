import React, { Component } from 'react';
import './HelloWorld.css';

class HelloWorld extends Component {
  constructor(props) {
    super(props);
    this.state = { greeting: 'Hello' };
    this.frenchify = this.frenchify.bind(this);
    this.removeGreeting = this.removeGreeting.bind(this);
  }
  frenchify() {
    this.setState({ greeting: 'Bonjour' });
  }
  removeGreeting() {
    this.props.removeGreeting(this.props.name);
  }
  render() {
    return (
      <div className="HelloWorld">
        {this.state.greeting} {this.props.name}!
        <br />
        <button onClick={this.frenchify}>Frenchify!</button>
        <br />
        <button onClick={this.removeGreeting}>Remove Me!</button>
      </div>
    );
  }
}

export default HelloWorld;
