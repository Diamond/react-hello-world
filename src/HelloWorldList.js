import React, { Component } from 'react';

// NOTE: Missing this from tutorial!
import HelloWorld from './HelloWorld';

import AddGreeter from './AddGreeter';

class HelloWorldList extends Component {
  constructor(props) {
    super(props);
    this.state = { greetings: ['Jim', 'Sally', 'Bender'] };
    this.addGreeting = this.addGreeting.bind(this);
    this.removeGreeting = this.removeGreeting.bind(this);
  }
  renderGreetings() {
    return this.state.greetings.map(name => (
      <HelloWorld
        key={name}
        name={name}
        removeGreeting={this.removeGreeting}
      />
    ))
  }
  addGreeting(newName) {
    this.setState({ greetings: [...this.state.greetings, newName] });
  }
  removeGreeting(removeName) {
    const filteredGreetings = this.state.greetings.filter(name => {
      return name !== removeName;
    });
    this.setState({ greetings: filteredGreetings });
  }
  render() {
    return (
      <div className="HelloWorldList">
        <AddGreeter addGreeting={this.addGreeting} />
        {this.renderGreetings()}
      </div>
    );
  }
}

// NOTE: Missing this from tutorial!
export default HelloWorldList;
