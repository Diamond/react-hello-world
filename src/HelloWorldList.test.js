import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

import HelloWorldList from './HelloWorldList';
import AddGreeter from './AddGreeter';
import HelloWorld from './HelloWorld';

describe(HelloWorldList, () => {
  const component = shallow(
    <HelloWorldList />
  );

  it('renders and matches our snapshot', () => {
    const component = renderer.create(
      <HelloWorldList />
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('contains an AddGreeter subcomponent', () => {
    expect(component.find(AddGreeter)).toHaveLength(1);
  });

  it('contains at least one HelloWorld subcomponent', () => {
    expect(component.find(HelloWorld).length).toBeGreaterThan(0);
  });
});
