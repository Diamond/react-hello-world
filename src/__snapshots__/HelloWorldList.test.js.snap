exports[`class HelloWorldList extends /* istanbul ignore next */_react.Component {
  constructor(props) {
    super(props);
    this.state = { greetings: ['Jim', 'Sally', 'Bender'] };
    this.addGreeting = this.addGreeting.bind(this);
    this.removeGreeting = this.removeGreeting.bind(this);
  }
  renderGreetings() {
    return this.state.greetings.map(name =>
    /* istanbul ignore next */_react2.default.createElement( /* istanbul ignore next */_HelloWorld2.default, /* istanbul ignore next */{
      key: name,
      name: name,
      removeGreeting: this.removeGreeting, __source: { fileName: _jsxFileName, lineNumber: 16 }, __self: this }));


  }
  addGreeting(newName) {
    this.setState({ greetings: [...this.state.greetings, newName] });
  }
  removeGreeting(removeName) {
    const filteredGreetings = this.state.greetings.filter(name => {
      return name !== removeName;
    });
    this.setState({ greetings: filteredGreetings });
  }
  render() {
    return (
      /* istanbul ignore next */_react2.default.createElement( /* istanbul ignore next */'div', /* istanbul ignore next */{ className: 'HelloWorldList', __source: { fileName: _jsxFileName, lineNumber: 34 }, __self: this },
        /* istanbul ignore next */_react2.default.createElement( /* istanbul ignore next */_AddGreeter2.default, /* istanbul ignore next */{ addGreeting: this.addGreeting, __source: { fileName: _jsxFileName, lineNumber: 35 }, __self: this }),
        this.renderGreetings()));


  }} renders and matches our snapshot 1`] = `
<div
  className="HelloWorldList">
  <div
    className="AddGreeter">
    <input
      onChange={[Function]}
      type="text"
      value="" />
      
    <button
      onClick={[Function]}>
      Add
    </button>
  </div>
  <div
    className="HelloWorld">
    Hello
     
    Jim
    !
    <br />
    <button
      className="frenchify"
      onClick={[Function]}>
      Frenchify!
    </button>
    <br />
    <button
      className="remove"
      onClick={[Function]}>
      Remove Me!
    </button>
  </div>
  <div
    className="HelloWorld">
    Hello
     
    Sally
    !
    <br />
    <button
      className="frenchify"
      onClick={[Function]}>
      Frenchify!
    </button>
    <br />
    <button
      className="remove"
      onClick={[Function]}>
      Remove Me!
    </button>
  </div>
  <div
    className="HelloWorld">
    Hello
     
    Bender
    !
    <br />
    <button
      className="frenchify"
      onClick={[Function]}>
      Frenchify!
    </button>
    <br />
    <button
      className="remove"
      onClick={[Function]}>
      Remove Me!
    </button>
  </div>
</div>
`;
